File: templates/example-interface.rb
#!/usr/bin/env ruby
# hello.rb - Simple greeting utility in Ruby
# RC Summary: Displays a customizable greeting message (Ruby version)

require 'optparse'

# Default options
options = {
  format: "Hello, NAME!",
  uppercase: false,
  name: "Friend"
}

# Parse command line options
parser = OptionParser.new do |opts|
  opts.banner = "Usage: hello [options] [name]"
  
  opts.on("--format FORMAT", "Greeting format (default: 'Hello, NAME!')") do |format|
    options[:format] = format
  end
  
  opts.on("-u", "--uppercase", "Convert to uppercase") do
    options[:uppercase] = true
  end
  
  opts.on("--summary", "Show summary for rc help") do
    puts "Displays a customizable greeting message (Ruby version)"
    exit 0
  end
  
  opts.on("--version", "Show version information") do
    puts "hello - rcForge Utility v0.4.1"
    exit 0
  end
  
  opts.on_tail("-h", "--help", "Show this help message") do
    puts opts
    exit 0
  end
end

# Parse arguments
begin
  parser.parse!
  
  # Get name from positional argument if provided
  options[:name] = ARGV[0] if ARGV[0]
  
  # Main functionality
  greeting = options[:format].gsub('NAME', options[:name])
  greeting.upcase! if options[:uppercase]
  
  puts greeting
  exit 0
rescue OptionParser::InvalidOption => e
  puts "Error: #{e}"
  puts parser
  exit 1
end

File: templates/example-interface.pl
#!/usr/bin/env perl
# hello.pl - Simple greeting utility in Perl
# RC Summary: Displays a customizable greeting message (Perl version)

use strict;
use warnings;
use Getopt::Long;
use Pod::Usage;

# Default values
my $format = "Hello, NAME!";
my $uppercase = 0;
my $show_summary = 0;
my $show_version = 0;
my $show_help = 0;
my $name = "Friend";

# Parse command line options
GetOptions(
    "format=s"  => \$format,
    "uppercase|u" => \$uppercase,
    "summary"   => \$show_summary,
    "version"   => \$show_version,
    "help|h"    => \$show_help
) or pod2usage(1);

# Get name from positional argument if provided
$name = $ARGV[0] if @ARGV > 0;

# Handle special rc command flags
if ($show_help) {
    pod2usage(-verbose => 2);
    exit 0;
}

if ($show_summary) {
    print "Displays a customizable greeting message (Perl version)\n";
    exit 0;
}

if ($show_version) {
    print "hello - rcForge Utility v0.4.1\n";
    exit 0;
}

# Main functionality
my $greeting = $format;
$greeting =~ s/NAME/$name/g;

if ($uppercase) {
    $greeting = uc($greeting);
}

print "$greeting\n";
exit 0;

__END__

=head1 NAME

hello - Displays a customizable greeting message

=head1 SYNOPSIS

hello [options] [name]

=head1 OPTIONS

=over 8

=item B<--format>=FORMAT

Greeting format (default: "Hello, NAME!")

=item B<--uppercase>, B<-u>

Convert greeting to uppercase

=item B<--summary>

Show summary for rc help

=item B<--version>

Show version information

=item B<--help>, B<-h>

Show this help message

=back

=head1 DESCRIPTION

A simple utility that displays a greeting message with configurable name and format.

=head1 EXAMPLES

hello World               # Outputs: Hello, World!
hello --format="Hi, NAME" Mark  # Outputs: Hi, Mark
hello --uppercase Alice        # Outputs: HELLO, ALICE!

=cut
